domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs

```diff
@@ -38,4 +41,44 @@ public async Task Create_ShouldFail_WhenModelInvalid()
         var response = await authenticatedClient.PostAsJsonAsync("api/transfer-agreements", new { });
         response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
     }
+
+    [Fact]
+    public async Task GetBySubjectId_ShouldReturnTransferAgreements_WhenUserHasTransferAgreements()
```

Comment:
For me, it appears that the integration tests and unit tests, test the same thing (they test one endpoint at a time and in isolation). Does that give much value having both?